### DISCUSSION TITLE
Crowdlab algorithm question

### DISCUSSION THREAD

**Discussion by vahuja4** [cleanlab/cleanlab#871] [Q&A] [answered]
From the crowdlab paper: "Illustrating how many real-world multiannotator datasets look, Figure 1 shows a disparity in annotator quality as well as many examples whose consensus
label will be incorrect if we rely on majority vote (nonetheless often done in practice due to its straightforward appeal).
**Unsurprisingly, consensus labels are more likely to be incorrect for those examples with fewer annotations.** An effective
method to estimate consensus label quality should properly
account for the number of annotations an example has received, as well as the quality of the annotators who selected
these labels". 
So, if we have 3 annotators annotating all the data, does the crowdlab algorithm still help with creating the consensus label as opposed to majority vote? Because there is no concept of fewer annotations in my dataset.

**Comment by huiwengoh** [comment, marked as answer]
Hi @vahuja4, thanks for the question!

Even if you had 3 annotators (or any constant number) annotating all the data, CROWDLAB **will still be helpful in providing more robust consensus labels** for your dataset. 

This is because in addition to the annotations from your annotators, CROWDLAB utilizes information (predictions) from a trained classifier to determine the best consensus label for each example. The CROWDLAB algorithm will automatically weigh the trustworthiness of the annotators and classifier and ensemble their annotations/predictions to provide you a consensus label, alongside a score for each example indicating the confidence level of that consensus label. 

This process is similar regardless if you have a constant or varying number of annotators for each example. In fact, Appendix C of the [CROWDLAB paper](https://arxiv.org/abs/2210.06812) showcases CROWDLAB in use with a dataset where each example was annotated by 50 annotators!

Lastly, you can use any trained classifier with CROWDLAB, but as with most of the cleanlab package, a better classifier would likely yield better results 🙂

**Reply by vahuja4** [reply]
@huiwengoh - thank you for your reply! I have one more question: for the crowdlab algorithm to work, we need the following:

1. a trained classifier (M)
2. we also need a dataset with the following features X(assuming 3 annotators): a1's label, a2's label, a3's label,  M's prediction. Here, Y is the ground_truth. We use the crowdlab algorithm using this dataset to figure out the ensemble weighting, etc.
3. finally, we need a hold-out set which has the same input features X but no ground truth and we use the crowdlab algorithm to give us the consensus label along the with score.

Can you please confirm?

**Reply by huiwengoh** [reply]
You are mostly on the right track! But one key details is that CROWDLAB does **not** require ground truth labels, in fact in most applications the ground truth labels will not be available to you. 

Rather, you can obtain out-of-sample predictions by your classifier for all your datapoints by performing cross validation, and then CROWDLAB can provide you with consensus label and scores for all those datapoints as well.

Here is a [quick tutorial](https://docs.cleanlab.ai/stable/tutorials/multiannotator.html) that might make that process clearer to you, hope it helps and please let us know if you have any other questions!
